{
  "nbformat": 4,
  "nbformat_minor": 0,
  "metadata": {
    "colab": {
      "provenance": [],
      "authorship_tag": "ABX9TyOrbkEgUfqkR9qnVmfJKTBK",
      "include_colab_link": true
    },
    "kernelspec": {
      "name": "python3",
      "display_name": "Python 3"
    },
    "language_info": {
      "name": "python"
    }
  },
  "cells": [
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "view-in-github",
        "colab_type": "text"
      },
      "source": [
        "<a href=\"https://colab.research.google.com/github/Voronga/dz/blob/main/dz2/Kallisto.ipynb\" target=\"_parent\"><img src=\"https://colab.research.google.com/assets/colab-badge.svg\" alt=\"Open In Colab\"/></a>"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 1,
      "metadata": {
        "id": "k5D6W9J0QTmX",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "1596810c-3c45-458c-e6e6-7a04751d0444"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Cloning into 'kallisto'...\n",
            "remote: Enumerating objects: 4124, done.\u001b[K\n",
            "remote: Counting objects: 100% (138/138), done.\u001b[K\n",
            "remote: Compressing objects: 100% (69/69), done.\u001b[K\n",
            "remote: Total 4124 (delta 70), reused 108 (delta 61), pack-reused 3986\u001b[K\n",
            "Receiving objects: 100% (4124/4124), 4.24 MiB | 17.44 MiB/s, done.\n",
            "Resolving deltas: 100% (2685/2685), done.\n",
            "Reading package lists... Done\n",
            "Building dependency tree       \n",
            "Reading state information... Done\n",
            "The following package was automatically installed and is no longer required:\n",
            "  libnvidia-common-460\n",
            "Use 'apt autoremove' to remove it.\n",
            "The following additional packages will be installed:\n",
            "  automake autotools-dev libsigsegv2 m4\n",
            "Suggested packages:\n",
            "  autoconf-archive gnu-standards autoconf-doc libtool gettext m4-doc\n",
            "The following NEW packages will be installed:\n",
            "  autoconf automake autotools-dev libsigsegv2 m4\n",
            "0 upgraded, 5 newly installed, 0 to remove and 20 not upgraded.\n",
            "Need to get 1,082 kB of archives.\n",
            "After this operation, 3,994 kB of additional disk space will be used.\n",
            "Get:1 http://archive.ubuntu.com/ubuntu bionic/main amd64 libsigsegv2 amd64 2.12-1 [14.7 kB]\n",
            "Get:2 http://archive.ubuntu.com/ubuntu bionic/main amd64 m4 amd64 1.4.18-1 [197 kB]\n",
            "Get:3 http://archive.ubuntu.com/ubuntu bionic/main amd64 autoconf all 2.69-11 [322 kB]\n",
            "Get:4 http://archive.ubuntu.com/ubuntu bionic/main amd64 autotools-dev all 20180224.1 [39.6 kB]\n",
            "Get:5 http://archive.ubuntu.com/ubuntu bionic/main amd64 automake all 1:1.15.1-3ubuntu2 [509 kB]\n",
            "Fetched 1,082 kB in 1s (1,015 kB/s)\n",
            "Selecting previously unselected package libsigsegv2:amd64.\n",
            "(Reading database ... 157604 files and directories currently installed.)\n",
            "Preparing to unpack .../libsigsegv2_2.12-1_amd64.deb ...\n",
            "Unpacking libsigsegv2:amd64 (2.12-1) ...\n",
            "Selecting previously unselected package m4.\n",
            "Preparing to unpack .../archives/m4_1.4.18-1_amd64.deb ...\n",
            "Unpacking m4 (1.4.18-1) ...\n",
            "Selecting previously unselected package autoconf.\n",
            "Preparing to unpack .../autoconf_2.69-11_all.deb ...\n",
            "Unpacking autoconf (2.69-11) ...\n",
            "Selecting previously unselected package autotools-dev.\n",
            "Preparing to unpack .../autotools-dev_20180224.1_all.deb ...\n",
            "Unpacking autotools-dev (20180224.1) ...\n",
            "Selecting previously unselected package automake.\n",
            "Preparing to unpack .../automake_1%3a1.15.1-3ubuntu2_all.deb ...\n",
            "Unpacking automake (1:1.15.1-3ubuntu2) ...\n",
            "Setting up libsigsegv2:amd64 (2.12-1) ...\n",
            "Setting up m4 (1.4.18-1) ...\n",
            "Setting up autotools-dev (20180224.1) ...\n",
            "Setting up autoconf (2.69-11) ...\n",
            "Setting up automake (1:1.15.1-3ubuntu2) ...\n",
            "update-alternatives: using /usr/bin/automake-1.15 to provide /usr/bin/automake (automake) in auto mode\n",
            "Processing triggers for libc-bin (2.27-3ubuntu1.5) ...\n",
            "Processing triggers for man-db (2.8.3-2ubuntu0.1) ...\n",
            "-- The C compiler identification is GNU 7.5.0\n",
            "-- The CXX compiler identification is GNU 7.5.0\n",
            "-- Detecting C compiler ABI info\n",
            "-- Detecting C compiler ABI info - done\n",
            "-- Check for working C compiler: /usr/bin/cc - skipped\n",
            "-- Detecting C compile features\n",
            "-- Detecting C compile features - done\n",
            "-- Detecting CXX compiler ABI info\n",
            "-- Detecting CXX compiler ABI info - done\n",
            "-- Check for working CXX compiler: /usr/bin/c++ - skipped\n",
            "-- Detecting CXX compile features\n",
            "-- Detecting CXX compile features - done\n",
            "\u001b[0mshared build\u001b[0m\n",
            "-- Looking for pthread.h\n",
            "-- Looking for pthread.h - found\n",
            "-- Performing Test CMAKE_HAVE_LIBC_PTHREAD\n",
            "-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Failed\n",
            "-- Looking for pthread_create in pthreads\n",
            "-- Looking for pthread_create in pthreads - not found\n",
            "-- Looking for pthread_create in pthread\n",
            "-- Looking for pthread_create in pthread - found\n",
            "-- Found Threads: TRUE  \n",
            "-- Found ZLIB: /usr/lib/x86_64-linux-gnu/libz.so (found version \"1.2.11\") \n",
            "-- Configuring done\n",
            "-- Generating done\n",
            "-- Build files have been written to: /content/kallisto/build\n",
            "[  3%] \u001b[34m\u001b[1mCreating directories for 'htslib'\u001b[0m\n",
            "[  6%] \u001b[34m\u001b[1mNo download step for 'htslib'\u001b[0m\n",
            "[ 10%] \u001b[34m\u001b[1mNo update step for 'htslib'\u001b[0m\n",
            "[ 13%] \u001b[34m\u001b[1mNo patch step for 'htslib'\u001b[0m\n",
            "[ 17%] \u001b[34m\u001b[1mPerforming configure step for 'htslib'\u001b[0m\n",
            "checking for gcc... gcc\n",
            "checking whether the C compiler works... yes\n",
            "checking for C compiler default output file name... a.out\n",
            "checking for suffix of executables... \n",
            "checking whether we are cross compiling... no\n",
            "checking for suffix of object files... o\n",
            "checking whether we are using the GNU C compiler... yes\n",
            "checking whether gcc accepts -g... yes\n",
            "checking for gcc option to accept ISO C89... none needed\n",
            "checking for ranlib... ranlib\n",
            "checking for special C compiler options needed for large files... no\n",
            "checking for _FILE_OFFSET_BITS value needed for large files... no\n",
            "checking shared library type... plain .so\n",
            "checking how to run the C preprocessor... gcc -E\n",
            "checking for grep that handles long lines and -e... /bin/grep\n",
            "checking for egrep... /bin/grep -E\n",
            "checking for ANSI C header files... yes\n",
            "checking for sys/types.h... yes\n",
            "checking for sys/stat.h... yes\n",
            "checking for stdlib.h... yes\n",
            "checking for string.h... yes\n",
            "checking for memory.h... yes\n",
            "checking for strings.h... yes\n",
            "checking for inttypes.h... yes\n",
            "checking for stdint.h... yes\n",
            "checking for unistd.h... yes\n",
            "checking for stdlib.h... (cached) yes\n",
            "checking for unistd.h... (cached) yes\n",
            "checking for sys/param.h... yes\n",
            "checking for getpagesize... yes\n",
            "checking for working mmap... yes\n",
            "checking for gmtime_r... yes\n",
            "checking whether fdatasync is declared... yes\n",
            "checking for fdatasync... yes\n",
            "checking for library containing log... -lm\n",
            "checking for zlib.h... yes\n",
            "checking for inflate in -lz... yes\n",
            "checking for library containing connect... none required\n",
            "configure: WARNING: GCS support not enabled: requires libcurl support\n",
            "configure: WARNING: S3 support not enabled: requires libcurl support\n",
            "configure: creating ./config.status\n",
            "config.status: creating config.mk\n",
            "config.status: creating htslib.pc.tmp\n",
            "config.status: creating config.h\n",
            "[ 20%] \u001b[34m\u001b[1mPerforming build step for 'htslib'\u001b[0m\n",
            "[ 24%] \u001b[34m\u001b[1mNo install step for 'htslib'\u001b[0m\n",
            "[ 27%] \u001b[34m\u001b[1mCompleted 'htslib'\u001b[0m\n",
            "[ 27%] Built target htslib\n",
            "[ 31%] \u001b[32mBuilding CXX object src/CMakeFiles/kallisto_core.dir/BUSData.cpp.o\u001b[0m\n",
            "[ 34%] \u001b[32mBuilding CXX object src/CMakeFiles/kallisto_core.dir/BUSTools.cpp.o\u001b[0m\n",
            "[ 37%] \u001b[32mBuilding CXX object src/CMakeFiles/kallisto_core.dir/Bootstrap.cpp.o\u001b[0m\n",
            "[ 41%] \u001b[32mBuilding CXX object src/CMakeFiles/kallisto_core.dir/GeneModel.cpp.o\u001b[0m\n",
            "[ 44%] \u001b[32mBuilding CXX object src/CMakeFiles/kallisto_core.dir/H5Writer.cpp.o\u001b[0m\n",
            "[ 48%] \u001b[32mBuilding CXX object src/CMakeFiles/kallisto_core.dir/Kmer.cpp.o\u001b[0m\n",
            "[ 51%] \u001b[32mBuilding CXX object src/CMakeFiles/kallisto_core.dir/KmerIndex.cpp.o\u001b[0m\n",
            "[ 55%] \u001b[32mBuilding CXX object src/CMakeFiles/kallisto_core.dir/KmerIterator.cpp.o\u001b[0m\n",
            "[ 58%] \u001b[32mBuilding CXX object src/CMakeFiles/kallisto_core.dir/Merge.cpp.o\u001b[0m\n",
            "[ 62%] \u001b[32mBuilding CXX object src/CMakeFiles/kallisto_core.dir/MinCollector.cpp.o\u001b[0m\n",
            "[ 65%] \u001b[32mBuilding CXX object src/CMakeFiles/kallisto_core.dir/PlaintextWriter.cpp.o\u001b[0m\n",
            "[ 68%] \u001b[32mBuilding CXX object src/CMakeFiles/kallisto_core.dir/ProcessReads.cpp.o\u001b[0m\n",
            "[ 72%] \u001b[32mBuilding CXX object src/CMakeFiles/kallisto_core.dir/PseudoBam.cpp.o\u001b[0m\n",
            "[ 75%] \u001b[32mBuilding CXX object src/CMakeFiles/kallisto_core.dir/common.cpp.o\u001b[0m\n",
            "[ 79%] \u001b[32mBuilding CXX object src/CMakeFiles/kallisto_core.dir/h5utils.cpp.o\u001b[0m\n",
            "[ 82%] \u001b[32mBuilding CXX object src/CMakeFiles/kallisto_core.dir/hash.cpp.o\u001b[0m\n",
            "[ 86%] \u001b[32mBuilding CXX object src/CMakeFiles/kallisto_core.dir/main.cpp.o\u001b[0m\n",
            "[ 89%] \u001b[32mBuilding CXX object src/CMakeFiles/kallisto_core.dir/weights.cpp.o\u001b[0m\n",
            "[ 93%] \u001b[32m\u001b[1mLinking CXX static library libkallisto_core.a\u001b[0m\n",
            "[ 93%] Built target kallisto_core\n",
            "[ 96%] \u001b[32mBuilding CXX object src/CMakeFiles/kallisto.dir/main.cpp.o\u001b[0m\n",
            "[100%] \u001b[32m\u001b[1mLinking CXX executable kallisto\u001b[0m\n",
            "[100%] Built target kallisto\n"
          ]
        }
      ],
      "source": [
        "!git clone https://github.com/pachterlab/kallisto.git\n",
        "!apt-get install autoconf\n",
        "!cd kallisto && mkdir build && cd build && cmake .. && make"
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!curl -fsSL https://get.nextflow.io | bash"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "oT7M4EGdNm_U",
        "outputId": "cb151703-b273-4392-c909-ca487b61e7b1"
      },
      "execution_count": 2,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "CAPSULE: Downloading dependency org.multiverse:multiverse-core:jar:0.7.0\n",
            "CAPSULE: Downloading dependency ch.qos.logback:logback-classic:jar:1.2.9\n",
            "CAPSULE: Downloading dependency org.checkerframework:checker-compat-qual:jar:2.0.0\n",
            "CAPSULE: Downloading dependency org.codehaus.mojo:animal-sniffer-annotations:jar:1.14\n",
            "CAPSULE: Downloading dependency com.google.errorprone:error_prone_annotations:jar:2.1.3\n",
            "CAPSULE: Downloading dependency org.codehaus.groovy:groovy-templates:jar:3.0.10\n",
            "CAPSULE: Downloading dependency com.google.guava:guava:jar:24.1.1-jre\n",
            "CAPSULE: Downloading dependency org.codehaus.jsr166-mirror:jsr166y:jar:1.7.0\n",
            "CAPSULE: Downloading dependency io.nextflow:nf-commons:jar:22.04.5\n",
            "CAPSULE: Downloading dependency org.slf4j:jcl-over-slf4j:jar:1.7.32\n",
            "CAPSULE: Downloading dependency com.beust:jcommander:jar:1.35\n",
            "CAPSULE: Downloading dependency org.slf4j:jul-to-slf4j:jar:1.7.32\n",
            "CAPSULE: Downloading dependency com.google.j2objc:j2objc-annotations:jar:1.1\n",
            "CAPSULE: Downloading dependency ch.grengine:grengine:jar:1.3.0\n",
            "CAPSULE: Downloading dependency jline:jline:jar:2.9\n",
            "CAPSULE: Downloading dependency com.github.zafarkhaja:java-semver:jar:0.9.0\n",
            "CAPSULE: Downloading dependency org.jsoup:jsoup:jar:1.14.3\n",
            "CAPSULE: Downloading dependency org.codehaus.groovy:groovy-xml:jar:3.0.10\n",
            "CAPSULE: Downloading dependency javax.mail:mail:jar:1.4.7\n",
            "CAPSULE: Downloading dependency io.nextflow:nf-httpfs:jar:22.04.5\n",
            "CAPSULE: Downloading dependency commons-lang:commons-lang:jar:2.6\n",
            "CAPSULE: Downloading dependency org.iq80.leveldb:leveldb-api:jar:0.12\n",
            "CAPSULE: Downloading dependency org.codehaus.gpars:gpars:jar:1.2.1\n",
            "CAPSULE: Downloading dependency org.slf4j:slf4j-api:jar:1.7.32\n",
            "CAPSULE: Downloading dependency com.google.code.findbugs:jsr305:jar:1.3.9\n",
            "CAPSULE: Downloading dependency org.objenesis:objenesis:jar:2.1\n",
            "CAPSULE: Downloading dependency com.esotericsoftware.kryo:kryo:jar:2.24.0\n",
            "CAPSULE: Downloading dependency org.apache.ivy:ivy:jar:2.3.0\n",
            "CAPSULE: Downloading dependency org.codehaus.groovy:groovy-json:jar:3.0.10\n",
            "CAPSULE: Downloading dependency com.googlecode.javaewah:JavaEWAH:jar:1.1.6\n",
            "CAPSULE: Downloading dependency org.codehaus.groovy:groovy-nio:jar:3.0.10\n",
            "CAPSULE: Downloading dependency org.codehaus.groovy:groovy:jar:3.0.10\n",
            "CAPSULE: Downloading dependency org.slf4j:log4j-over-slf4j:jar:1.7.32\n",
            "CAPSULE: Downloading dependency dev.failsafe:failsafe:jar:3.1.0\n",
            "CAPSULE: Downloading dependency javax.activation:activation:jar:1.1.1\n",
            "CAPSULE: Downloading dependency org.yaml:snakeyaml:jar:1.28\n",
            "CAPSULE: Downloading dependency com.jcraft:jsch:jar:0.1.54\n",
            "CAPSULE: Downloading dependency org.pf4j:pf4j:jar:3.4.1\n",
            "CAPSULE: Downloading dependency org.iq80.leveldb:leveldb:jar:0.12\n",
            "CAPSULE: Downloading dependency ch.qos.logback:logback-core:jar:1.2.9\n",
            "CAPSULE: Downloading dependency com.jcraft:jzlib:jar:1.1.1\n",
            "CAPSULE: Downloading dependency com.google.code.gson:gson:jar:2.2.4\n",
            "CAPSULE: Downloading dependency io.nextflow:nextflow:jar:22.04.5\n",
            "CAPSULE: Downloading dependency org.eclipse.jgit:org.eclipse.jgit:jar:5.2.1.201812262042-r\n",
            "CAPSULE: Downloading dependency org.pf4j:pf4j-update:jar:2.3.0\n",
            "CAPSULE: Downloading dependency commons-codec:commons-codec:jar:1.10\n",
            "\n",
            "      N E X T F L O W\n",
            "      version 22.04.5 build 5708\n",
            "      created 15-07-2022 16:09 UTC \n",
            "      cite doi:10.1038/nbt.3820\n",
            "      http://nextflow.io\n",
            "\n",
            "\n",
            "Nextflow installation completed. Please note:\n",
            "- the executable file `nextflow` has been created in the folder: /content\n",
            "- you may complete the installation by moving it to a directory in your $PATH\n",
            "\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!wget https://ftp-trace.ncbi.nlm.nih.gov/sra/sdk/3.0.0/sratoolkit.3.0.0-ubuntu64.tar.gz\n",
        "!tar -xvzf sratoolkit.3.0.0-ubuntu64.tar.gz"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "2L-1No7aOJOO",
        "outputId": "0794b202-7991-4c22-bc2d-5739a45293c2"
      },
      "execution_count": 3,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "--2022-09-27 09:50:15--  https://ftp-trace.ncbi.nlm.nih.gov/sra/sdk/3.0.0/sratoolkit.3.0.0-ubuntu64.tar.gz\n",
            "Resolving ftp-trace.ncbi.nlm.nih.gov (ftp-trace.ncbi.nlm.nih.gov)... 165.112.9.230, 130.14.250.7, 2607:f220:41e:250::11, ...\n",
            "Connecting to ftp-trace.ncbi.nlm.nih.gov (ftp-trace.ncbi.nlm.nih.gov)|165.112.9.230|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 85777118 (82M) [application/x-gzip]\n",
            "Saving to: ‘sratoolkit.3.0.0-ubuntu64.tar.gz’\n",
            "\n",
            "sratoolkit.3.0.0-ub 100%[===================>]  81.80M  95.5MB/s    in 0.9s    \n",
            "\n",
            "2022-09-27 09:50:16 (95.5 MB/s) - ‘sratoolkit.3.0.0-ubuntu64.tar.gz’ saved [85777118/85777118]\n",
            "\n",
            "sratoolkit.3.0.0-ubuntu64/\n",
            "sratoolkit.3.0.0-ubuntu64/bin/\n",
            "sratoolkit.3.0.0-ubuntu64/bin/abi-dump\n",
            "sratoolkit.3.0.0-ubuntu64/bin/abi-dump.3\n",
            "sratoolkit.3.0.0-ubuntu64/bin/abi-load\n",
            "sratoolkit.3.0.0-ubuntu64/bin/abi-load.3\n",
            "sratoolkit.3.0.0-ubuntu64/bin/align-info\n",
            "sratoolkit.3.0.0-ubuntu64/bin/align-info.3\n",
            "sratoolkit.3.0.0-ubuntu64/bin/bam-load\n",
            "sratoolkit.3.0.0-ubuntu64/bin/bam-load.3\n",
            "sratoolkit.3.0.0-ubuntu64/bin/cache-mgr\n",
            "sratoolkit.3.0.0-ubuntu64/bin/cache-mgr.3\n",
            "sratoolkit.3.0.0-ubuntu64/bin/cg-load\n",
            "sratoolkit.3.0.0-ubuntu64/bin/cg-load.3\n",
            "sratoolkit.3.0.0-ubuntu64/bin/dump-ref-fasta\n",
            "sratoolkit.3.0.0-ubuntu64/bin/dump-ref-fasta.3\n",
            "sratoolkit.3.0.0-ubuntu64/bin/fasterq-dump\n",
            "sratoolkit.3.0.0-ubuntu64/bin/fasterq-dump.3\n",
            "sratoolkit.3.0.0-ubuntu64/bin/fasterq-dump.3.0.0\n",
            "sratoolkit.3.0.0-ubuntu64/bin/fastq-dump\n",
            "sratoolkit.3.0.0-ubuntu64/bin/fastq-dump.3\n",
            "sratoolkit.3.0.0-ubuntu64/bin/fastq-dump.3.0.0\n",
            "sratoolkit.3.0.0-ubuntu64/bin/fastq-load\n",
            "sratoolkit.3.0.0-ubuntu64/bin/fastq-load.3\n",
            "sratoolkit.3.0.0-ubuntu64/bin/helicos-load\n",
            "sratoolkit.3.0.0-ubuntu64/bin/helicos-load.3\n",
            "sratoolkit.3.0.0-ubuntu64/bin/illumina-dump\n",
            "sratoolkit.3.0.0-ubuntu64/bin/illumina-dump.3\n",
            "sratoolkit.3.0.0-ubuntu64/bin/illumina-load\n",
            "sratoolkit.3.0.0-ubuntu64/bin/illumina-load.3\n",
            "sratoolkit.3.0.0-ubuntu64/bin/kar\n",
            "sratoolkit.3.0.0-ubuntu64/bin/kar.3\n",
            "sratoolkit.3.0.0-ubuntu64/bin/kdbmeta\n",
            "sratoolkit.3.0.0-ubuntu64/bin/kdbmeta.3\n",
            "sratoolkit.3.0.0-ubuntu64/bin/latf-load\n",
            "sratoolkit.3.0.0-ubuntu64/bin/latf-load.3\n",
            "sratoolkit.3.0.0-ubuntu64/bin/pacbio-load\n",
            "sratoolkit.3.0.0-ubuntu64/bin/pacbio-load.3\n",
            "sratoolkit.3.0.0-ubuntu64/bin/prefetch\n",
            "sratoolkit.3.0.0-ubuntu64/bin/prefetch.3\n",
            "sratoolkit.3.0.0-ubuntu64/bin/prefetch.3.0.0\n",
            "sratoolkit.3.0.0-ubuntu64/bin/rcexplain\n",
            "sratoolkit.3.0.0-ubuntu64/bin/rcexplain.3\n",
            "sratoolkit.3.0.0-ubuntu64/bin/sam-dump\n",
            "sratoolkit.3.0.0-ubuntu64/bin/sam-dump.3\n",
            "sratoolkit.3.0.0-ubuntu64/bin/sam-dump.3.0.0\n",
            "sratoolkit.3.0.0-ubuntu64/bin/sff-dump\n",
            "sratoolkit.3.0.0-ubuntu64/bin/sff-dump.3\n",
            "sratoolkit.3.0.0-ubuntu64/bin/sff-load\n",
            "sratoolkit.3.0.0-ubuntu64/bin/sff-load.3\n",
            "sratoolkit.3.0.0-ubuntu64/bin/sra-pileup\n",
            "sratoolkit.3.0.0-ubuntu64/bin/sra-pileup.3\n",
            "sratoolkit.3.0.0-ubuntu64/bin/sra-pileup.3.0.0\n",
            "sratoolkit.3.0.0-ubuntu64/bin/sra-search\n",
            "sratoolkit.3.0.0-ubuntu64/bin/sra-search.3\n",
            "sratoolkit.3.0.0-ubuntu64/bin/sra-sort\n",
            "sratoolkit.3.0.0-ubuntu64/bin/sra-sort-cg\n",
            "sratoolkit.3.0.0-ubuntu64/bin/sra-sort-cg.3\n",
            "sratoolkit.3.0.0-ubuntu64/bin/sra-sort.3\n",
            "sratoolkit.3.0.0-ubuntu64/bin/sra-stat\n",
            "sratoolkit.3.0.0-ubuntu64/bin/sra-stat.3\n",
            "sratoolkit.3.0.0-ubuntu64/bin/srapath\n",
            "sratoolkit.3.0.0-ubuntu64/bin/srapath.3\n",
            "sratoolkit.3.0.0-ubuntu64/bin/srapath.3.0.0\n",
            "sratoolkit.3.0.0-ubuntu64/bin/srf-load\n",
            "sratoolkit.3.0.0-ubuntu64/bin/srf-load.3\n",
            "sratoolkit.3.0.0-ubuntu64/bin/test-sra\n",
            "sratoolkit.3.0.0-ubuntu64/bin/test-sra.3\n",
            "sratoolkit.3.0.0-ubuntu64/bin/vdb-config\n",
            "sratoolkit.3.0.0-ubuntu64/bin/vdb-config.3\n",
            "sratoolkit.3.0.0-ubuntu64/bin/vdb-copy\n",
            "sratoolkit.3.0.0-ubuntu64/bin/vdb-copy.3\n",
            "sratoolkit.3.0.0-ubuntu64/bin/vdb-decrypt\n",
            "sratoolkit.3.0.0-ubuntu64/bin/vdb-decrypt.3\n",
            "sratoolkit.3.0.0-ubuntu64/bin/vdb-dump\n",
            "sratoolkit.3.0.0-ubuntu64/bin/vdb-dump.3\n",
            "sratoolkit.3.0.0-ubuntu64/bin/vdb-dump.3.0.0\n",
            "sratoolkit.3.0.0-ubuntu64/bin/vdb-encrypt\n",
            "sratoolkit.3.0.0-ubuntu64/bin/vdb-encrypt.3\n",
            "sratoolkit.3.0.0-ubuntu64/bin/vdb-lock\n",
            "sratoolkit.3.0.0-ubuntu64/bin/vdb-lock.3\n",
            "sratoolkit.3.0.0-ubuntu64/bin/vdb-unlock\n",
            "sratoolkit.3.0.0-ubuntu64/bin/vdb-unlock.3\n",
            "sratoolkit.3.0.0-ubuntu64/bin/vdb-validate\n",
            "sratoolkit.3.0.0-ubuntu64/bin/vdb-validate.3\n",
            "sratoolkit.3.0.0-ubuntu64/bin/ncbi/\n",
            "sratoolkit.3.0.0-ubuntu64/bin/ncbi/certs.kfg\n",
            "sratoolkit.3.0.0-ubuntu64/bin/ncbi/default.kfg\n",
            "sratoolkit.3.0.0-ubuntu64/bin/ncbi/vdb-copy.kfg\n",
            "sratoolkit.3.0.0-ubuntu64/bin/srapath-orig.3.0.0\n",
            "sratoolkit.3.0.0-ubuntu64/bin/sratools.3.0.0\n",
            "sratoolkit.3.0.0-ubuntu64/bin/srf-load.3.0.0\n",
            "sratoolkit.3.0.0-ubuntu64/bin/test-sra.3.0.0\n",
            "sratoolkit.3.0.0-ubuntu64/bin/vdb-config.3.0.0\n",
            "sratoolkit.3.0.0-ubuntu64/bin/vdb-copy.3.0.0\n",
            "sratoolkit.3.0.0-ubuntu64/bin/vdb-decrypt.3.0.0\n",
            "sratoolkit.3.0.0-ubuntu64/bin/vdb-dump-orig.3.0.0\n",
            "sratoolkit.3.0.0-ubuntu64/bin/vdb-encrypt.3.0.0\n",
            "sratoolkit.3.0.0-ubuntu64/bin/vdb-lock.3.0.0\n",
            "sratoolkit.3.0.0-ubuntu64/bin/vdb-unlock.3.0.0\n",
            "sratoolkit.3.0.0-ubuntu64/bin/vdb-validate.3.0.0\n",
            "sratoolkit.3.0.0-ubuntu64/bin/abi-dump.3.0.0\n",
            "sratoolkit.3.0.0-ubuntu64/bin/abi-load.3.0.0\n",
            "sratoolkit.3.0.0-ubuntu64/bin/align-info.3.0.0\n",
            "sratoolkit.3.0.0-ubuntu64/bin/bam-load.3.0.0\n",
            "sratoolkit.3.0.0-ubuntu64/bin/cache-mgr.3.0.0\n",
            "sratoolkit.3.0.0-ubuntu64/bin/cg-load.3.0.0\n",
            "sratoolkit.3.0.0-ubuntu64/bin/dump-ref-fasta.3.0.0\n",
            "sratoolkit.3.0.0-ubuntu64/bin/fasterq-dump-orig.3.0.0\n",
            "sratoolkit.3.0.0-ubuntu64/bin/fastq-dump-orig.3.0.0\n",
            "sratoolkit.3.0.0-ubuntu64/bin/fastq-load.3.0.0\n",
            "sratoolkit.3.0.0-ubuntu64/bin/helicos-load.3.0.0\n",
            "sratoolkit.3.0.0-ubuntu64/bin/illumina-dump.3.0.0\n",
            "sratoolkit.3.0.0-ubuntu64/bin/illumina-load.3.0.0\n",
            "sratoolkit.3.0.0-ubuntu64/bin/kar.3.0.0\n",
            "sratoolkit.3.0.0-ubuntu64/bin/kdbmeta.3.0.0\n",
            "sratoolkit.3.0.0-ubuntu64/bin/latf-load.3.0.0\n",
            "sratoolkit.3.0.0-ubuntu64/bin/pacbio-load.3.0.0\n",
            "sratoolkit.3.0.0-ubuntu64/bin/prefetch-orig.3.0.0\n",
            "sratoolkit.3.0.0-ubuntu64/bin/rcexplain.3.0.0\n",
            "sratoolkit.3.0.0-ubuntu64/bin/sam-dump-orig.3.0.0\n",
            "sratoolkit.3.0.0-ubuntu64/bin/sff-dump.3.0.0\n",
            "sratoolkit.3.0.0-ubuntu64/bin/sff-load.3.0.0\n",
            "sratoolkit.3.0.0-ubuntu64/bin/sra-pileup-orig.3.0.0\n",
            "sratoolkit.3.0.0-ubuntu64/bin/sra-search.3.0.0\n",
            "sratoolkit.3.0.0-ubuntu64/bin/sra-sort-cg.3.0.0\n",
            "sratoolkit.3.0.0-ubuntu64/bin/sra-sort.3.0.0\n",
            "sratoolkit.3.0.0-ubuntu64/bin/sra-stat.3.0.0\n",
            "sratoolkit.3.0.0-ubuntu64/example/\n",
            "sratoolkit.3.0.0-ubuntu64/example/perl/\n",
            "sratoolkit.3.0.0-ubuntu64/example/perl/base-stats.pl\n",
            "sratoolkit.3.0.0-ubuntu64/example/perl/dump-reference.pl\n",
            "sratoolkit.3.0.0-ubuntu64/example/perl/gene-lookup.pl\n",
            "sratoolkit.3.0.0-ubuntu64/example/perl/mismatch-stats.pl\n",
            "sratoolkit.3.0.0-ubuntu64/example/perl/quality-stats.pl\n",
            "sratoolkit.3.0.0-ubuntu64/example/perl/simplefastq.pl\n",
            "sratoolkit.3.0.0-ubuntu64/example/perl/splitfastq.pl\n",
            "sratoolkit.3.0.0-ubuntu64/README-blastn\n",
            "sratoolkit.3.0.0-ubuntu64/README-vdb-config\n",
            "sratoolkit.3.0.0-ubuntu64/schema/\n",
            "sratoolkit.3.0.0-ubuntu64/schema/align/\n",
            "sratoolkit.3.0.0-ubuntu64/schema/align/align.vschema\n",
            "sratoolkit.3.0.0-ubuntu64/schema/align/mate-cache.vschema\n",
            "sratoolkit.3.0.0-ubuntu64/schema/align/pileup-stats.vschema\n",
            "sratoolkit.3.0.0-ubuntu64/schema/align/qstat.vschema\n",
            "sratoolkit.3.0.0-ubuntu64/schema/align/refseq.vschema\n",
            "sratoolkit.3.0.0-ubuntu64/schema/align/seq.vschema\n",
            "sratoolkit.3.0.0-ubuntu64/schema/csra2/\n",
            "sratoolkit.3.0.0-ubuntu64/schema/csra2/csra2.vschema\n",
            "sratoolkit.3.0.0-ubuntu64/schema/csra2/read.vschema\n",
            "sratoolkit.3.0.0-ubuntu64/schema/csra2/reference.vschema\n",
            "sratoolkit.3.0.0-ubuntu64/schema/csra2/stats.vschema\n",
            "sratoolkit.3.0.0-ubuntu64/schema/insdc/\n",
            "sratoolkit.3.0.0-ubuntu64/schema/insdc/insdc.vschema\n",
            "sratoolkit.3.0.0-ubuntu64/schema/insdc/seq.vschema\n",
            "sratoolkit.3.0.0-ubuntu64/schema/insdc/sra.vschema\n",
            "sratoolkit.3.0.0-ubuntu64/schema/ncbi/\n",
            "sratoolkit.3.0.0-ubuntu64/schema/ncbi/clip.vschema\n",
            "sratoolkit.3.0.0-ubuntu64/schema/ncbi/ncbi.vschema\n",
            "sratoolkit.3.0.0-ubuntu64/schema/ncbi/pnbrdb.vschema\n",
            "sratoolkit.3.0.0-ubuntu64/schema/ncbi/seq-graph.vschema\n",
            "sratoolkit.3.0.0-ubuntu64/schema/ncbi/seq.vschema\n",
            "sratoolkit.3.0.0-ubuntu64/schema/ncbi/spotname.vschema\n",
            "sratoolkit.3.0.0-ubuntu64/schema/ncbi/sra.vschema\n",
            "sratoolkit.3.0.0-ubuntu64/schema/ncbi/stats.vschema\n",
            "sratoolkit.3.0.0-ubuntu64/schema/ncbi/trace.vschema\n",
            "sratoolkit.3.0.0-ubuntu64/schema/ncbi/varloc.vschema\n",
            "sratoolkit.3.0.0-ubuntu64/schema/ncbi/wgs-contig.vschema\n",
            "sratoolkit.3.0.0-ubuntu64/schema/sra/\n",
            "sratoolkit.3.0.0-ubuntu64/schema/sra/454.vschema\n",
            "sratoolkit.3.0.0-ubuntu64/schema/sra/abi.vschema\n",
            "sratoolkit.3.0.0-ubuntu64/schema/sra/generic-fastq.vschema\n",
            "sratoolkit.3.0.0-ubuntu64/schema/sra/helicos.vschema\n",
            "sratoolkit.3.0.0-ubuntu64/schema/sra/illumina.vschema\n",
            "sratoolkit.3.0.0-ubuntu64/schema/sra/ion-torrent.vschema\n",
            "sratoolkit.3.0.0-ubuntu64/schema/sra/nanopore.vschema\n",
            "sratoolkit.3.0.0-ubuntu64/schema/sra/pacbio.vschema\n",
            "sratoolkit.3.0.0-ubuntu64/schema/sra/pevents.vschema\n",
            "sratoolkit.3.0.0-ubuntu64/schema/vdb/\n",
            "sratoolkit.3.0.0-ubuntu64/schema/vdb/built-in.vschema\n",
            "sratoolkit.3.0.0-ubuntu64/schema/vdb/vdb.vschema\n",
            "sratoolkit.3.0.0-ubuntu64/CHANGES\n",
            "sratoolkit.3.0.0-ubuntu64/README.md\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!sratoolkit.3.0.0-ubuntu64/bin/vdb-config -i"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "mVpFDDu_OWl6",
        "outputId": "6ab9ff08-d4e5-4f0c-da7b-5dd6134cc832"
      },
      "execution_count": 4,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "\u001b[2J\u001b[?25l\u001b[?1000h\u001b[?1002h2022-09-27T09:50:23 vdb-config.3.0.0 fatal: SIGNAL - Segmentation fault \n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!./nextflow run my_pipeline.nf --SRA SRR6410603,SRR6410604 -resume"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "wLvG3F03PFwe",
        "outputId": "22a2ed7f-ac57-43fe-aaeb-29ad8fa36710"
      },
      "execution_count": 14,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "N E X T F L O W  ~  version 22.04.5\n",
            "Launching `my_pipeline.nf` [insane_mclean] DSL2 - revision: e95712f4ae\n",
            "\n",
            " Q U A L I T Y C O N T R O L \n",
            "=================================\n",
            "SRA number : [SRR6410603, SRR6410604]\n",
            "Results location : results/\n",
            "[-        ] process > DownloadFastQ -\u001b[K\n",
            "\u001b[2A\n",
            "[-        ] process > DownloadFastQ -\u001b[K\n",
            "[-        ] process > kallisto      -\u001b[K\n",
            "\u001b[3A\n",
            "[06/b3c5b2] process > DownloadFastQ (1) [100%] 2 of 2, cached: 2 ✔\u001b[K\n",
            "[-        ] process > kallisto          [  0%] 0 of 1\u001b[K\n",
            "\u001b[3A\n",
            "executor >  local (1)\u001b[K\n",
            "[06/b3c5b2] process > DownloadFastQ (1) [100%] 2 of 2, cached: 2 ✔\u001b[K\n",
            "[b1/d7ce90] process > kallisto          [  0%] 0 of 1\u001b[K\n",
            "\u001b[4A\n",
            "executor >  local (1)\u001b[K\n",
            "[06/b3c5b2] process > DownloadFastQ (1) [100%] 2 of 2, cached: 2 ✔\u001b[K\n",
            "[b1/d7ce90] process > kallisto          [  0%] 0 of 1\u001b[K\n",
            "\u001b[31mError executing process > 'kallisto'\u001b[K\n",
            "\u001b[K\n",
            "Caused by:\u001b[K\n",
            "  Process `kallisto` terminated with an error exit status (137)\u001b[K\n",
            "\u001b[K\n",
            "Command executed:\u001b[K\n",
            "\u001b[K\n",
            "  mkdir kllst\u001b[K\n",
            "  wget https://ftp.ncbi.nlm.nih.gov/refseq/H_sapiens/annotation/GRCh38_latest/refseq_identifiers/GRCh38_latest_genomic.fna.gz\u001b[K\n",
            "  gunzip GRCh38_latest_genomic.fna.gz\u001b[K\n",
            "  cp GRCh38_latest_genomic.fna GRCh38_latest_genomic.fasta\u001b[K\n",
            "  /content/kallisto/build/src/kallisto index -i GRCh38_latest_genomic.idx GRCh38_latest_genomic.fasta\u001b[K\n",
            "  /content/kallisto/build/src/kallisto quant -i GRCh38_latest_genomic.idx -o results x\u001b[K\n",
            "\u001b[K\n",
            "Command exit status:\u001b[K\n",
            "  137\u001b[K\n",
            "\u001b[K\n",
            "Command output:\u001b[K\n",
            "  (empty)\u001b[K\n",
            "\u001b[K\n",
            "Command error:\u001b[K\n",
            "  948200K .......... .......... .......... .......... .......... 99% 8.52M 0s\u001b[K\n",
            "  948250K .......... .......... .......... .......... .......... 99% 15.8M 0s\u001b[K\n",
            "  948300K .......... .......... .......... .......... .......... 99% 19.5M 0s\u001b[K\n",
            "  948350K .......... .......... .......... .......... .......... 99% 44.1M 0s\u001b[K\n",
            "  948400K .......... .......... .......... .......... .......... 99% 36.4M 0s\u001b[K\n",
            "  948450K .......... .......... .......... .......... .......... 99% 44.4M 0s\u001b[K\n",
            "  948500K .......... .......... .......... .......... .......... 99% 9.84M 0s\u001b[K\n",
            "  948550K .......... .......... .......... .......... .......... 99% 8.86M 0s\u001b[K\n",
            "  948600K .......... .......... .......... .......... .......... 99% 6.61M 0s\u001b[K\n",
            "  948650K .......... .......... .......... .......... .......... 99% 56.6M 0s\u001b[K\n",
            "  948700K .......... .......... .......... .......... .......... 99% 60.6M 0s\u001b[K\n",
            "  948750K .......... .......... .......... .......... .......... 99% 7.47M 0s\u001b[K\n",
            "  948800K .......... .......... .......... .......... .......... 99% 9.11M 0s\u001b[K\n",
            "  948850K .......... .......... .......... .......... .......... 99% 11.0M 0s\u001b[K\n",
            "  948900K .......... .......... .......... .......... .......... 99% 16.8M 0s\u001b[K\n",
            "  948950K .......... .......... .......... .......... .......... 99% 54.1M 0s\u001b[K\n",
            "  949000K .......... .......... .......... .......... .......... 99% 46.1M 0s\u001b[K\n",
            "  949050K .......... .......... .......... .......... .......... 99% 60.1M 0s\u001b[K\n",
            "  949100K .......... .......... .......... .......... .......... 99% 15.2M 0s\u001b[K\n",
            "  949150K .......... .......... .......... .......... .......... 99% 9.08M 0s\u001b[K\n",
            "  949200K .......... .......... .......... .......... .......... 99% 7.80M 0s\u001b[K\n",
            "  949250K .......... .......... .......... .......... .......... 99% 6.50M 0s\u001b[K\n",
            "  949300K .......... .......... .......... .......... .......... 99% 42.1M 0s\u001b[K\n",
            "  949350K .......... .......... .......... .......... .......... 99% 3.74M 0s\u001b[K\n",
            "  949400K .......... .......... .......... .......... .......... 99% 5.77M 0s\u001b[K\n",
            "  949450K .......... .......... .......... .......... .......... 99% 7.69M 0s\u001b[K\n",
            "  949500K .......... .......... .......... .......... .......... 99% 2.48M 0s\u001b[K\n",
            "  949550K .......... .......... .......... .......... .......... 99% 14.7M 0s\u001b[K\n",
            "  949600K .......... .......... .......... .......... .......... 99% 5.33M 0s\u001b[K\n",
            "  949650K .......... .......... .......... .......... .......... 99% 5.90M 0s\u001b[K\n",
            "  949700K .......... .......... .......... .......... .......... 99% 30.7M 0s\u001b[K\n",
            "  949750K .......... .......... .......... .......... .......... 99% 29.4M 0s\u001b[K\n",
            "  949800K .......... .......... .......... .......... .......... 99% 4.84M 0s\u001b[K\n",
            "  949850K .......... .......... .......... .......... .......... 99% 4.72M 0s\u001b[K\n",
            "  949900K .......... .......... .......... .......... .......... 99% 10.0M 0s\u001b[K\n",
            "  949950K .......... .......... .......... .......... .......... 99% 43.4M 0s\u001b[K\n",
            "  950000K .......... .......... .......... .......... .......... 99% 37.3M 0s\u001b[K\n",
            "  950050K .......... .......... .......... .......... ......    100% 10.8M=10s\u001b[K\n",
            "  \u001b[K\n",
            "  2022-09-27 12:46:55 (92.2 MB/s) - ‘GRCh38_latest_genomic.fna.gz’ saved [972898531/972898531]\u001b[K\n",
            "  \u001b[K\n",
            "  \u001b[K\n",
            "  [build] loading fasta file GRCh38_latest_genomic.fasta\u001b[K\n",
            "  [build] k-mer length: 31\u001b[K\n",
            "  [build] warning: replaced 161611482 non-ACGUT characters in the input sequence\u001b[K\n",
            "          with pseudorandom nucleotides\u001b[K\n",
            "  [build] counting k-mers ... tcmalloc: large alloc 1610612736 bytes == 0x5596b596c000 @  0x7fdaebd8d887 0x55957fe5982d 0x55957fe6140b 0x55957fe23f40 0x7fdaeac0dc87 0x55957fe2c10a\u001b[K\n",
            "  tcmalloc: large alloc 3221225472 bytes == 0x55971596c000 @  0x7fdaebd8d887 0x55957fe5982d 0x55957fe6140b 0x55957fe23f40 0x7fdaeac0dc87 0x55957fe2c10a\u001b[K\n",
            "  tcmalloc: large alloc 6442450944 bytes == 0x5597d619e000 @  0x7fdaebd8d887 0x55957fe5982d 0x55957fe6140b 0x55957fe23f40 0x7fdaeac0dc87 0x55957fe2c10a\u001b[K\n",
            "  .command.sh: line 6:  3511 Killed                  /content/kallisto/build/src/kallisto index -i GRCh38_latest_genomic.idx GRCh38_latest_genomic.fasta\u001b[K\n",
            "\u001b[K\n",
            "Work dir:\u001b[K\n",
            "  /content/work/b1/d7ce9073bf537fffce982e2c315bc2\u001b[K\n",
            "\u001b[K\n",
            "Tip: you can replicate the issue by changing to the process work dir and entering the command `bash .command.run`\u001b[K\n",
            "\u001b[K\n",
            "\u001b[39m\u001b[K\n",
            "\u001b[82A\n",
            "executor >  local (1)\u001b[K\n",
            "[06/b3c5b2] process > DownloadFastQ (1) [100%] 2 of 2, cached: 2 ✔\u001b[K\n",
            "[b1/d7ce90] process > kallisto          [100%] 1 of 1, failed: 1 ✘\u001b[K\n",
            "\u001b[31mError executing process > 'kallisto'\u001b[K\n",
            "\u001b[K\n",
            "Caused by:\u001b[K\n",
            "  Process `kallisto` terminated with an error exit status (137)\u001b[K\n",
            "\u001b[K\n",
            "Command executed:\u001b[K\n",
            "\u001b[K\n",
            "  mkdir kllst\u001b[K\n",
            "  wget https://ftp.ncbi.nlm.nih.gov/refseq/H_sapiens/annotation/GRCh38_latest/refseq_identifiers/GRCh38_latest_genomic.fna.gz\u001b[K\n",
            "  gunzip GRCh38_latest_genomic.fna.gz\u001b[K\n",
            "  cp GRCh38_latest_genomic.fna GRCh38_latest_genomic.fasta\u001b[K\n",
            "  /content/kallisto/build/src/kallisto index -i GRCh38_latest_genomic.idx GRCh38_latest_genomic.fasta\u001b[K\n",
            "  /content/kallisto/build/src/kallisto quant -i GRCh38_latest_genomic.idx -o results x\u001b[K\n",
            "\u001b[K\n",
            "Command exit status:\u001b[K\n",
            "  137\u001b[K\n",
            "\u001b[K\n",
            "Command output:\u001b[K\n",
            "  (empty)\u001b[K\n",
            "\u001b[K\n",
            "Command error:\u001b[K\n",
            "  948200K .......... .......... .......... .......... .......... 99% 8.52M 0s\u001b[K\n",
            "  948250K .......... .......... .......... .......... .......... 99% 15.8M 0s\u001b[K\n",
            "  948300K .......... .......... .......... .......... .......... 99% 19.5M 0s\u001b[K\n",
            "  948350K .......... .......... .......... .......... .......... 99% 44.1M 0s\u001b[K\n",
            "  948400K .......... .......... .......... .......... .......... 99% 36.4M 0s\u001b[K\n",
            "  948450K .......... .......... .......... .......... .......... 99% 44.4M 0s\u001b[K\n",
            "  948500K .......... .......... .......... .......... .......... 99% 9.84M 0s\u001b[K\n",
            "  948550K .......... .......... .......... .......... .......... 99% 8.86M 0s\u001b[K\n",
            "  948600K .......... .......... .......... .......... .......... 99% 6.61M 0s\u001b[K\n",
            "  948650K .......... .......... .......... .......... .......... 99% 56.6M 0s\u001b[K\n",
            "  948700K .......... .......... .......... .......... .......... 99% 60.6M 0s\u001b[K\n",
            "  948750K .......... .......... .......... .......... .......... 99% 7.47M 0s\u001b[K\n",
            "  948800K .......... .......... .......... .......... .......... 99% 9.11M 0s\u001b[K\n",
            "  948850K .......... .......... .......... .......... .......... 99% 11.0M 0s\u001b[K\n",
            "  948900K .......... .......... .......... .......... .......... 99% 16.8M 0s\u001b[K\n",
            "  948950K .......... .......... .......... .......... .......... 99% 54.1M 0s\u001b[K\n",
            "  949000K .......... .......... .......... .......... .......... 99% 46.1M 0s\u001b[K\n",
            "  949050K .......... .......... .......... .......... .......... 99% 60.1M 0s\u001b[K\n",
            "  949100K .......... .......... .......... .......... .......... 99% 15.2M 0s\u001b[K\n",
            "  949150K .......... .......... .......... .......... .......... 99% 9.08M 0s\u001b[K\n",
            "  949200K .......... .......... .......... .......... .......... 99% 7.80M 0s\u001b[K\n",
            "  949250K .......... .......... .......... .......... .......... 99% 6.50M 0s\u001b[K\n",
            "  949300K .......... .......... .......... .......... .......... 99% 42.1M 0s\u001b[K\n",
            "  949350K .......... .......... .......... .......... .......... 99% 3.74M 0s\u001b[K\n",
            "  949400K .......... .......... .......... .......... .......... 99% 5.77M 0s\u001b[K\n",
            "  949450K .......... .......... .......... .......... .......... 99% 7.69M 0s\u001b[K\n",
            "  949500K .......... .......... .......... .......... .......... 99% 2.48M 0s\u001b[K\n",
            "  949550K .......... .......... .......... .......... .......... 99% 14.7M 0s\u001b[K\n",
            "  949600K .......... .......... .......... .......... .......... 99% 5.33M 0s\u001b[K\n",
            "  949650K .......... .......... .......... .......... .......... 99% 5.90M 0s\u001b[K\n",
            "  949700K .......... .......... .......... .......... .......... 99% 30.7M 0s\u001b[K\n",
            "  949750K .......... .......... .......... .......... .......... 99% 29.4M 0s\u001b[K\n",
            "  949800K .......... .......... .......... .......... .......... 99% 4.84M 0s\u001b[K\n",
            "  949850K .......... .......... .......... .......... .......... 99% 4.72M 0s\u001b[K\n",
            "  949900K .......... .......... .......... .......... .......... 99% 10.0M 0s\u001b[K\n",
            "  949950K .......... .......... .......... .......... .......... 99% 43.4M 0s\u001b[K\n",
            "  950000K .......... .......... .......... .......... .......... 99% 37.3M 0s\u001b[K\n",
            "  950050K .......... .......... .......... .......... ......    100% 10.8M=10s\u001b[K\n",
            "  \u001b[K\n",
            "  2022-09-27 12:46:55 (92.2 MB/s) - ‘GRCh38_latest_genomic.fna.gz’ saved [972898531/972898531]\u001b[K\n",
            "  \u001b[K\n",
            "  \u001b[K\n",
            "  [build] loading fasta file GRCh38_latest_genomic.fasta\u001b[K\n",
            "  [build] k-mer length: 31\u001b[K\n",
            "  [build] warning: replaced 161611482 non-ACGUT characters in the input sequence\u001b[K\n",
            "          with pseudorandom nucleotides\u001b[K\n",
            "  [build] counting k-mers ... tcmalloc: large alloc 1610612736 bytes == 0x5596b596c000 @  0x7fdaebd8d887 0x55957fe5982d 0x55957fe6140b 0x55957fe23f40 0x7fdaeac0dc87 0x55957fe2c10a\u001b[K\n",
            "  tcmalloc: large alloc 3221225472 bytes == 0x55971596c000 @  0x7fdaebd8d887 0x55957fe5982d 0x55957fe6140b 0x55957fe23f40 0x7fdaeac0dc87 0x55957fe2c10a\u001b[K\n",
            "  tcmalloc: large alloc 6442450944 bytes == 0x5597d619e000 @  0x7fdaebd8d887 0x55957fe5982d 0x55957fe6140b 0x55957fe23f40 0x7fdaeac0dc87 0x55957fe2c10a\u001b[K\n",
            "  .command.sh: line 6:  3511 Killed                  /content/kallisto/build/src/kallisto index -i GRCh38_latest_genomic.idx GRCh38_latest_genomic.fasta\u001b[K\n",
            "\u001b[K\n",
            "Work dir:\u001b[K\n",
            "  /content/work/b1/d7ce9073bf537fffce982e2c315bc2\u001b[K\n",
            "\u001b[K\n",
            "Tip: you can replicate the issue by changing to the process work dir and entering the command `bash .command.run`\u001b[K\n",
            "\u001b[K\n",
            "\u001b[39m\u001b[K\n",
            "\n"
          ]
        }
      ]
    }
  ]
}